### ISSUE TITLE
System.NullReferenceException not serializable.

### ISSUE THREAD

**Issue by PencilBow** [dotnet/orleans#3772]
Hi.

When an System.NullReferenceException (or System.Exception) is thrown by a grain:
```
    [StorageProvider(ProviderName = "MyProvider")]
    public class HelloGrain : Grain, IHelloGrain {
        async public Task ThrowNullReference() {
            List<string> l = null;
            l.Clear();
        }
    }
```
` grain.ThrowNullReference().Wait();`

The client receives the following error:

```
Unhandled Exception: System.AggregateException: One or more errors occurred. (No copier found for object of type System.NullReferenceException. Perhaps you need to mark it [Serializable] or define a custom serializer for it?) ---> Orleans.Runtime.OrleansException: No copier found for object of type System.NullReferenceException. Perhaps you need to mark it [Serializable] or define a custom serializer for it?
   at Orleans.Serialization.SerializationManager.DeepCopierHelper(Type t, Object original, ICopyContext context)
   at Orleans.Serialization.SerializationManager.DeepCopyInner(Object original, ICopyContext context)
   at Orleans.Serialization.SerializationManager.DeepCopy(Object original)
   at Orleans.Runtime.InsideRuntimeClient.SafeSendExceptionResponse(Message message, Exception ex)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Program.Main(String[] args) in 
<snip>\Program.cs:line 29
```

However, when I throw a self-made Exception, I do get a stacktrace:
```
[Serializable]
public class DeadHostException : Exception {
    public DeadHostException() { }
    public DeadHostException(string message) : base(message) { }
    public DeadHostException(string message, Exception inner) : base(message, inner) { }
    protected DeadHostException(
        System.Runtime.Serialization.SerializationInfo info,
        System.Runtime.Serialization.StreamingContext context) : base(info, context) { }
}
```

```
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Program.Main(String[] args) in 
<snip>\Program.cs:line 26
DeadHostException: description of host dead host is
   at GrainCollection.HelloGrain.<ThrowError>d__1.MoveNext() in 
<snip>\HelloGrain.cs:line 30
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Generated8D53E870C11CC47.OrleansCodeGenHelloGrainMethodInvoker.<Invoke>d__0.MoveNext() in 
<snip>\GrainInterfaces.orleans.g.cs:line 118
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.Runtime.GrainMethodInvoker.<Invoke>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.Runtime.InsideRuntimeClient.<Invoke>d__70.MoveNext()
```

If I add `config.FallbackSerializationProvider = typeof(ILBasedSerializer).GetTypeInfo();` to the client-config, system exception get handled but my custom exception dont...
I've gone through #2528, #2633, #3066 but none of that solved the issue.

client-config:
```
        var config = new ClientConfiguration {
            DeploymentId = deploymentId,
            GatewayProvider = ClientConfiguration.GatewayProviderType.Config,
            GatewayListRefreshPeriod = TimeSpan.FromSeconds(2)
        };
        config.Gateways.Add(new IPEndPoint(IPAddress.Loopback, port));

        this._client = new ClientBuilder()
                    .AddApplicationPartsFromReferences(typeof(IHelloGrain).Assembly)
                    .AddApplicationPartsFromReferences(typeof(Program).Assembly) // client-assembly

                    .UseConfiguration(config)
                    .ConfigureLogging(logging => logging.AddProvider(new ConsoleLoggerProvider((text, logLevel) => logLevel >= LogLevel.Warning, true)))
                    .Build();

         await this._client.Connect();
```

silo-config:
```
            var config = ClusterConfiguration.LocalhostPrimarySilo(port1, port2);
            config.AddMemoryStorageProvider("MyProvider");
            config.AddMemoryStorageProvider("PubSubStore");

            var host = new SiloHostBuilder()
                .AddApplicationPartsFromReferences(typeof(IHelloGrain).Assembly)
                .AddApplicationPartsFromReferences(typeof(HelloGrain).Assembly)
                .UseConfiguration(config)
                .ConfigureLogging(logging => logging.AddProvider(new ConsoleLoggerProvider((text, logLevel) => logLevel >= LogLevel.Warning, true)))
                .Build();
```

Windows 10 (Pro, 64bit), using Orleans 2.0.0 beta-1 (NuGet) on all projects. 

This is my first look at Orleans, so please forgive any noob mistakes.

**Comment by dVakulen** [comment]
It should've been fixed by https://github.com/dotnet/orleans/pull/3659. If you can not use latest master - the second 2.0 beta will be out soon. 

cc @ReubenBond (its not grafted yet)

**Comment by PencilBow** [comment]
@dVakulen  thanks for the quick response.

I have tried all permutations with the fallbackserialization (host & client, host-only, client-only), without succes. Is there an ETA for the the second beta?

**Comment by dVakulen** [comment]
@PencilBow I'd expect it to be released on the next week.

**Comment by ReubenBond** [comment]
@PencilBow could you try a nightly build from https://dotnet.myget.org/F/orleans-ci/api/v3/index.json ?

**Comment by PencilBow** [comment]
@ReubenBond I'm slightly confused as to where the builds are? that link points me to a json with 20ish links, and I cant make out from the descriptions where I need to go

**Comment by ReubenBond** [comment]
This should be fixed by #3794
